### PR TITLE
DDF-3060 changed to extend jetty's impl instead of the abstract class to avoid a hack in pax web

### PR DESCRIPTION
**Pull request by stustison**
… to avoid a hack in pax web

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@jlcsmith @beyelerb @millerw8 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Security](https://github.com/orgs/codice/teams/security)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@stustison
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

File: platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/HashSessionIdManager.java [hunk 1]
```diff
@@ -47,13 +47,14 @@
...
 import org.eclipse.jetty.server.session.AbstractSession;
-import org.eclipse.jetty.server.session.AbstractSessionIdManager;
 
 /**
  * Custom implementation of the {@link org.eclipse.jetty.server.SessionIdManager} that shares session
  * data between sessions in a cluster.
  */
-public class HashSessionIdManager extends AbstractSessionIdManager {
+@edu.umd.cs.findbugs.annotations.SuppressWarnings(value = "NM_SAME_SIMPLE_NAME_AS_SUPERCLASS", justification = "not changing")
+public class HashSessionIdManager extends org.eclipse.jetty.server.session.HashSessionIdManager {
```
Comment (coyotesqrl) [inline platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/HashSessionIdManager.java]: Sonarqube doesn't like it either. :/

Is a class rename out of the question? If not, we'll need to mark this as a non-issue for Sonarqube as well.

File: platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/HashSessionIdManager.java [hunk 2]
```diff
@@ -47,13 +47,14 @@
...
+    //changed this to extend Jetty's impl instead of the abstract class because of a hack in pax web
```
Comment (millerw8) [inline platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/HashSessionIdManager.java]: I'm a little confused, this seems to be going from a jetty abstract class to a jetty class. Where does the pax web hack come in?  Also, What's the difference between this class and the jetty version of HashSessionIdManager.  It seems we override all of its functions anyway, so the fact that we are using it over AbstractSessionIdManager doesn't change anything.  And that class seems to be the same as this class except this uses a concurrentHashMap and that class uses synchronized everywhere.

Comment (stustison) [inline platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/HashSessionIdManager.java]: Pax Web is expecting an instance of the Jetty implementation, which ours was not. That ended up skipping a bunch of code that we actually need. There are also additional changes to our id manager that aren't in the Jetty Impl. This is what we use to hook in SSO.

File: platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/HashSessionIdManager.java (hunk 1 above)
Comment (stustison) [inline platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/HashSessionIdManager.java]: I really had no desire to change the class name.

**Comment by stustison** [comment]
I just changed the name because I was tired of fighting all the static analysis tools.

**Comment by stustison** [comment]
build now

**Pull request merged**